doc/performance_blog_post/performance_blog_post.md

```diff
@@ -0,0 +1,109 @@
+
+![Performance test results of parquet file loading 1](slowParquetTestResults1.png)
+![Performance test results of parquet file loading 2](slowParquetTestResults2.png)
+
+In the charts it becomes clear that the Virtual Schema at that point has an issue with loading data from few big files. When loading from more files, it loads the same amount of data a lot quicker.
+
+Starting with this analysis, we discussed with our product management and decided to invest into performance improvements for loading big parquet files. More precisely we set the goal to make improvements so that the `testLoadFewBigSalesParquetFiles` also runs in less than 250s.
+
+## Improving the Performance
+
+Next we analyzed the source code for possible reasons for the slow reading of big parquet files. There we identified the following issues:
+
+* Unequal distribution
+* Too few parallelization for few parquet files
+
+The query processing of the Files Virtual Schemas consists of two phases: A planning phase and an execution phase. In the planning phase the adapter analyzes the query and starts multiple workers (UDFs) that then run the execution phase. For technical reasons the workers can not communicate at the moment. So the Virtual Schema parallelizes the import job over multiple workers.
+
+### Distributing the File
+
+Until now this parallelization was realized using hash binning. So each worker received a range and then loaded files that names hash values fell into that range. That approach is great for distributing lots of files, since it does not require to list all files in advance. Doing so could cause memory overflow when distributing for example 1,000,000 files with long names.
+
+However, for only a few files, this works rather bad:
+Imagine distributing 4 files over 4 workers. A good distribution is obvious here. Each worker receives on file. The workers then load the files in parallel. With hash binning it's however quite likely, that one worker receives two files and another none. This will increase the time for loading.
+
+![Example of file distribution](fileDistribution.png)
+
+To fix this issue we changed the implementation to distribute smaller amounts of files explicitly and only use hash binning for larger amounts.
+
+### Splitting Large Files
+
+The second problem occurs, if there are 4 workers but only one file. In that case only one worker will be used. To mitigate this we decided to split the file into multiple parts. That's not possible for all file formats. For example with JSON files where each file contains one document / row. For parquet, however it is. Parquet files contain multiple rows and group these rows into blocks (row groups). We changed our reader so that it's capable of only reading specific row groups. By that, we can again distribute the work over all available workers.
```

Comment:
```suggestion
The second problem occurs, if there are 4 workers but only one file. In that case only one worker will be used. To mitigate this issue, we decided to split the file into multiple parts. That's not possible for all file formats. For example with JSON files where each file contains one document or one row. For parquet, however it is possible. Parquet files contain multiple rows and group these rows into blocks called `row groups`. We changed our Parquet reader so that it's capable of only reading specific row groups. By that, we can again distribute the work over all available workers.
```